src/recorder/jamcontroller.cpp

```diff
@@ -77,15 +78,25 @@ void CJamController::SetRecordingDir ( QString newRecordingDir,
                                        int     iServerFrameSizeSamples,
                                        bool    bDisableRecording )
 {
-    if ( bRecorderInitialised && pthJamRecorder != nullptr )
+    if ( bRecorderInitialised )
     {
-        // We have a thread and we want to start a new one.
-        // We only want one running.
-        // This could take time, unfortunately.
-        // Hopefully changing recording directory will NOT happen during a long jam...
-        emit EndRecorderThread();
-        pthJamRecorder->wait();
-        pthJamRecorder = nullptr;
+        if ( pthJamRecorder != nullptr )
+        {
+            // We have a thread and we want to start a new one.
+            // We only want one running.
+            // This could take time, unfortunately.
+            // Hopefully changing recording directory will NOT happen during a long jam...
+            emit EndRecorderThread();
+            pthJamRecorder->wait();
+            delete pthJamRecorder;
+            pthJamRecorder = nullptr;
+        }
+        if ( pJamRecorder != nullptr )
+        {
+            // We have a recorder running already. Terminate it.
+            delete pJamRecorder;
+            pJamRecorder = nullptr;
+        }
     }
 
     if ( !newRecordingDir.isEmpty() )
```

Comment:
So inside this block, on the next line, check `if (pJamRecorder != null) { delete pJamRecorder; pJamRecorder = nullptr; }`.

Reply:
(The CJamRecorder instance is long running: normally for the duration of the Jamulus main process.)

Reply:
Ok, that's fine, I can move it to there. I'll check the detail tomorrow.